### PR TITLE
Add pre-push hook and composer test command

### PR DESCRIPTION
**Pull request by nidble**
Refs #53 

Implementing `composer test` and githook prepush integration. Commands parameters and settings taken from `circle.yml`. :)

### CODE REVIEW

File: etc/hooks/pre-push [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+# Final result
+echo ""
+
+if [ $canPush -eq 0 ]
+then
+    echo "[OK] Your code will be pushed young Padawan"
+    exit 0
+else
+    echo "[KO] Cancelling push due to test code style error (run 'composer test' for more information)"
```
Comment (JavierCane) [inline etc/hooks/pre-push]: Suggestion: replace `due to test code style error` by something less specific (we're not just checking out the code style, but actually running the acceptance and unit tests).

We could say something like `due to 'composer test' task error`. What do you think?

🙌

File: phpunit.xml.dist [hunk 2]
```diff
@@ -15,11 +15,11 @@
         convertNoticesToExceptions="true"
         convertWarningsToExceptions="true"
         processIsolation="false"
-        stopOnError="false"
-        stopOnFailure="false"
+        stopOnError="true"
+        stopOnFailure="true"
```
Comment (JavierCane) [inline phpunit.xml.dist]: Is it needed to change this policy?

The downside of this change if I'm not mistaken would be that we will not get the full report of the different tests that are failing. What are the benefits of this change?

Thanks!

File: etc/hooks/pre-push (hunk 1 above)
Comment (nidble) [reply etc/hooks/pre-push]: Good suggestion!

I'll fix this with: `[KO] Cancelling push due to 'composer test' task error (run 'composer test' for more information)`

File: phpunit.xml.dist (hunk 2 above)
Comment (nidble) [reply phpunit.xml.dist]: Yup,
without those fixes phpunit always exit with "0". Normally this is an acceptable behavior, but if I need prevent push by checking exit code, I need to impose onFailure and onError like the before snippet.

But feel free to suggest me the best approach do you prefer! :)

Comment (nidble) [reply phpunit.xml.dist]: A further consideration: instead to change dist file, I can improve phpunit like that:
`phpunit --stop-on-failure --stop-on-error [...]`

Comment (JavierCane) [reply phpunit.xml.dist]: Makes sense. Thanks for the clarification 🙂

Yes, I would vote for leaving the `stopOnError="false"` and `stopOnFailure="false"` config options in the `phpunit.xml.dist`. Why:

* This way we would be having a default config for PhpUnit useful for the majority of the use cases (locally developing and being interested in all the failing test due to our changes)
* We can cover the minority of the use cases (running CI pipeline or `pre-push` hooks) with the second approach you've proposed (running PhpUnit with the command line flags)

Do you agree? What do you think?

Comment (nidble) [reply phpunit.xml.dist]: Yep,

I do agree with you, I consider prepush like so a "safe guard" of pushing new code. Usually, during code session, I tend to run singular test more. After when I'm confortable with my code I start to run all test suite...

Pheew! Do you think this PR can be merged?  👼

**Pull request merged**